ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/messenger/data/realtime/RealtimeUserHelperTest.java

```diff
@@ -0,0 +1,160 @@
+import com.kayako.sdk.android.k5.core.KayakoLogHelper;
+import com.kayako.sdk.android.k5.kre.base.KreConnection;
+import com.kayako.sdk.android.k5.kre.base.KreConnectionFactory;
+import com.kayako.sdk.android.k5.kre.base.KreSubscription.OnSubscriptionListener;
+import com.kayako.sdk.android.k5.kre.base.credentials.KreCredentials;
+import com.kayako.sdk.android.k5.kre.base.user.KreUserSubscription;
+import com.kayako.sdk.android.k5.kre.helpers.RawUserOnlinePresenceListener;
+import com.kayako.sdk.android.k5.messenger.data.realtime.RealtimeUserHelper.UserPresenceListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({
+        MessengerOpenTracker.class,
+        KreStarterFactory.class,
+        KreUserSubscription.class,
+        KreConnectionFactory.class,
+        RealtimeUserHelper.class,
+        KayakoLogHelper.class,
+        Handler.class,
+        TextUtils.class
+})
+public class RealtimeUserHelperTest {
+    private static final String USER_PRESENCE_CHANNEL = "userPresenceChannel";
+    private static final long USER_ID = 1;
+
+    @Mock
+    private UserPresenceListener userPresenceListener;
+    @Mock
+    private KreStarter kreStarter;
+    @Mock
+    private KreUserSubscription kreUserSubscription;
+    @Mock
+    private KreConnection kreConnection;
+    @Captor
+    private ArgumentCaptor<OnSubscriptionListener> subscriptionListenerCaptor;
+    @Captor
+    private ArgumentCaptor<RawUserOnlinePresenceListener> onlinePresenceListenerCaptor;
+    @Captor
+    private ArgumentCaptor<Runnable> runnableCaptor;
+    @Captor
+    private ArgumentCaptor<Long> userIdCaptor;
+    @Mock
+    private Handler handler;
+
+    @Before
+    public void setUp() throws Exception {
+        mockStatic(MessengerOpenTracker.class);
+        mockStatic(KreStarterFactory.class);
+        mockStatic(TextUtils.class);
+        mockStatic(KreConnectionFactory.class);
+        mockStatic(KayakoLogHelper.class);
+
+        when(KreConnectionFactory.getConnection(false)).thenReturn(kreConnection);
+        whenNew(KreUserSubscription.class).withArguments(kreConnection, USER_PRESENCE_CHANNEL).thenReturn(kreUserSubscription);
+        whenNew(Handler.class).withNoArguments().thenReturn(handler);
+        when(KreStarterFactory.getKreStarterValues()).thenReturn(kreStarter);
+    }
+
+    @After
+    public void teatDown() {
+        RealtimeUserHelper.closeAll();
+    }
+
+    private void verifyCallbackCommon(boolean online) {
+        verify(kreUserSubscription).subscribe(any(KreCredentials.class), eq(USER_PRESENCE_CHANNEL), eq(USER_ID), subscriptionListenerCaptor.capture());
```

Comment:
⚠️  A line of code should not exceed 120 characters in order to be readable and avoid horizontal scroll. Consider using the 'line-wrapping' technique to format this line of code in multiple shorter lines. The formula for wrapping can be found at the below link.  

**[More details](https://confluence.devfactory.com/x/wgbhEw)**
          
## The possible solution in this case

```JAVA
        verify(kreUserSubscription).subscribe(any(KreCredentials.class), eq(USER_PRESENCE_CHANNEL),
                                              eq(USER_ID), subscriptionListenerCaptor.capture());

```
 

<a id="33387c50-4ee4-11e8-b617-bd6867d59d97" href="none" codereview-uuid-reserved-tag />